src/cdomains/intDomain.ml

```diff
@@ -779,6 +804,66 @@ struct
     in
     QCheck.(set_shrink shrink @@ set_print show @@ map (*~rev:BatOption.get*) (of_interval ik) pair_arb)
   let relift x = x
+
+  let abs x = let r = Ints_t.neg x
+   in if Ints_t.compare x r < 0 then r else x
+
+  let modulo n k =
+    let result = Ints_t.rem n k in
+    if Ints_t.compare result Ints_t.zero >= 0 then result
+    else Ints_t.add result  k
+
+  let refine_with_congruence ik (intv : t) (cong : (int_t * int_t ) option) : t =
+    match intv, cong with
+    | Some (x, y), Some (c, m) ->
+        if (m = Ints_t.zero) && (Ints_t.compare c x < 0 || Ints_t.compare c y > 0) then None
+        else if m = Ints_t.zero then Some (c, c)
```

Comment:
Should probably use `Ints_t.equal` instead of `=` since other comparisons are delegated.